### ISSUE TITLE
Feature Request: SVG icon support

### ISSUE THREAD

**Issue by davidklebanoff** [kartik-v/bootstrap-star-rating#91]
I'd love to see support for SVG icons as an alternative to font icons. Not sure if that's something you're interested in. It doesn't even necessarily need to be SVG specific, just a way to insert some raw html in place of the icons.

Example:

```html
<svg class="icon icon-star"><use xlink:href="#icon-star"></use></svg>
```

**Comment by kartik-v** [comment]
Currently CSS content embed is used for rendering the background and highlighting the icons. This can work with font icons because you just need to embed a character. Not sure how it would work with SVG.

**Comment by davidklebanoff** [comment]
I've played around with the idea of shoving the svg in the content directly, but I've hit a bit of a road block.

It works when inserted into the css directly:
```css
content: url('data:image/svg+xml,%3Csvg%20version%3D%271.1%27%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20xmlns%3Axlink%3D%27http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%27%20width%3D%2732%27%20height%3D%2732%27%20viewBox%3D%270%200%2032%2032%27%3E%3Cpath%20d%3D%27M20.6%2011l-4.6-10.5-4.6%2010.5h-10.8l7.8%207.9-3%2012.1%2010.6-6%2010.6%206-3-12.1%207.8-7.9z%27%3E%3C%2Fpath%3E%3C%2Fsvg%3E')
```

which embeds an encoded version of:

```svg
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="32" height="32" viewBox="0 0 32 32">
<path d="M20.6 11l-4.6-10.5-4.6 10.5h-10.8l7.8 7.9-3 12.1 10.6-6 10.6 6-3-12.1 7.8-7.9z"></path>
</svg>
```

But I haven't gotten it to work indirectly through the `content: attr(data-content)`. I also haven't looked into having multiple svgs in a single content block.

...I'm not sure this approach is worth it.

**Comment by kartik-v** [comment]
Using a SVG or an IMAGE requires a different way of handling. It can work for whole stars but may not be however as dynamic and flexible as the current solution (where you can add as many stars as you want, select any fractional star rating, change the symbols on the fly, change the sizes on the fly etc.)

**Comment by davidklebanoff** [comment]
Looking forward to see what you come up with.

I have abandoned the idea embedding the SVG within the content attribute, as it seems `content: attr(data-content)` is not currently supported with such contents.

**Comment by kartik-v** [comment]
I have done this and will be posting shortly. It required a bit rehaul of the overall code (including various BC breaking enhancements)

But I feel the end result is more extensible, more configurable, and will work across both the FONT icons as well as SVG space. 

An hint of the changes to come is highlighted in issue #92.

**Comment by kartik-v** [comment]
Resolved via MAJOR UPDATE of plugin to release v4.0.0. 

v4.x is a rewrite of the earlier 3.x version and has BC breaking changes. Look into the change log and documentation for all updates and the new THEME properties

You can look into a [demo of bootstrap-rating for SVG icon usage with Krajee SVG theme](http://plugins.krajee.com/star-rating-demo-theme-svg).
